src/AWS.Distro.OpenTelemetry.AutoInstrumentation/OtlpUdpExporter.cs

```diff
@@ -0,0 +1,343 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+using System.Diagnostics;
+using System.Net.Sockets;
+using System.Reflection;
+using System.Text;
+using Google.Protobuf;
+using Newtonsoft.Json;
+using OpenTelemetry;
+using OpenTelemetry.Proto.Collector.Trace.V1;
+using OpenTelemetry.Proto.Trace.V1;
+using OpenTelemetry.Resources;
+using OtlpResource = OpenTelemetry.Proto.Resource.V1;
+
+/// <summary>
+/// OTLP UDP Exporter class. This class is used to build an OtlpUdpExporter to registered as in exporter
+/// during the instrumentation initialization phase
+/// </summary>
+public class OtlpUdpExporter : BaseExporter<Activity>
+{
+    private UdpExporter udpExporter;
+    private string signalPrefix;
+    private Resource processResource;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="OtlpUdpExporter"/> class.
+    /// </summary>
+    /// <param name="endpoint">Endpoint to export requests to</param>
+    /// <param name="signalPrefix">Sampled vs UnSampled signal prefix</param>
+    /// <param name="processResource">Otel Resource object</param>
+    public OtlpUdpExporter(Resource processResource, string? endpoint = null, string? signalPrefix = null)
+    {
+        endpoint = endpoint ?? UdpExporter.DefaultEndpoint;
+        this.udpExporter = new UdpExporter(endpoint);
+        this.signalPrefix = signalPrefix ?? UdpExporter.DefaultFormatOtelTracesBinaryPrefix;
+        this.processResource = processResource;
+    }
+
+    /// <inheritdoc/>
+    public override ExportResult Export(in Batch<Activity> batch)
+    {
+        byte[]? serializedData = this.SerializeSpans(batch);
+        if (serializedData == null)
+        {
+            return ExportResult.Failure;
+        }
+
+        try
+        {
+            this.udpExporter.SendData(serializedData, this.signalPrefix);
+            return ExportResult.Success;
+        }
+        catch (Exception ex)
+        {
+            Console.WriteLine($"Error exporting spans: {ex.Message}");
+            return ExportResult.Failure;
+        }
+    }
+
+    /// <inheritdoc/>
+    protected override bool OnShutdown(int timeoutMilliseconds)
+    {
+        try
+        {
+            this.udpExporter.Shutdown();
```

Comment:
Does it support force flush when shutdown?

Reply:
Yes. In the plugin, we are adding the exporter as follows: `new BatchActivityExportProcessor(new OtlpUdpExporter())` and then we are adding the BatchActivityExportProcessor to the list of processors in the TracerProvider (according to otel spec). When the application is shutdown, BatchActivityExportProcessor.ForceFlush is called (forceflush is called on all registered processors) and that forceflush in upstream calls export on any remaining activities before shutting down.